integration-site/action.yml

```diff
@@ -0,0 +1,42 @@
+name: "Integration Site"
+description: "Site using Maven"
+
+#inputs:
+
+runs:
+  using: "composite"
+  steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+
+      - name: Set up Java 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          server-id: github
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+          restore-keys: v1-dependencies-
+
+      - name: Owasp cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository/org/owasp/dependency-check-data
+          key: owasp-cve-cache-{{ checksum "/tmp/owasp-cve-cache-date" }}
+          restore-keys: owasp-cve-cache-
+
+      - name: Maven site
+        shell: bash
+        run: mvn -s $GITHUB_WORKSPACE/.github/settings.xml site -DskipTests
+
+      - name: Upload site reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: site-reports
+          path: target/site
+
```

Comment:
For further reviews: artifacts can't be found until the whole actions workflow is completed. It seems to be a feature https://github.com/actions/upload-artifact/issues/53